### ISSUE TITLE
Every time I open the app again, I am signed out, why ?

### ISSUE THREAD

**Issue by mariuspena** [aws-amplify/aws-sdk-android#2528]
**State your question**
Every time I open the app again, I am signed out, why ?

**Which AWS Services are you utilizing?**
AWS cognito

**Provide code snippets (if applicable)**
I init Amplify in the Application.onCreate 
`RxAmplify.configure(applicationContext)
`
I check the session with
`RxAmplify.Auth.fetchAuthSession()
`-> every time I open the app it returns false

I sign int with 
`RxAmplify.Auth.signIn(username, password)
`-> I receive a AuthSignInResult where isSignInComplete is TRUE ! (everything seems to be good in the result) 
BUT in the console I get this silent error :

`D/AWSMobileClient: hasFederatedToken: false provider: cognito-idp.eu-west-1.amazonaws.com/eu-west-1_eudw64F6W
W/AWSMobileClient: Failed to federate tokens during sign-in
    java.lang.RuntimeException: Error in federating the token.
        at com.amazonaws.mobile.client.AWSMobileClient$10.run(AWSMobileClient.java:1803)
        at com.amazonaws.mobile.client.internal.InternalCallback.await(InternalCallback.java:115)
        at com.amazonaws.mobile.client.AWSMobileClient.federatedSignInWithoutAssigningState(AWSMobileClient.java:1731)
        at com.amazonaws.mobile.client.AWSMobileClient$6$1.onSuccess(AWSMobileClient.java:1223)
        at com.amazonaws.mobileconnectors.cognitoidentityprovider.CognitoUser.getSession(CognitoUser.java:1020)
        at com.amazonaws.mobile.client.AWSMobileClient$6.run(AWSMobileClient.java:1208)
        at com.amazonaws.mobile.client.internal.InternalCallback$1.run(InternalCallback.java:101)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: com.amazonaws.AmazonServiceException: 1 validation error detected: Value 'eu-west-1_eudw64F6W' at 'identityPoolId' failed to satisfy constraint: Member must satisfy regular expression pattern: [\w-]+:[0-9a-f-]+ (Service: AmazonCognitoIdentity; Status Code: 400; Error Code: ValidationException; Request ID: 9bef8fcd-5fb2-45ce-b160-8ed8954ca8e2)
        at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:742)
        at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:420)
        at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:229)
        at com.amazonaws.services.cognitoidentity.AmazonCognitoIdentityClient.invoke(AmazonCognitoIdentityClient.java:1836)
        at com.amazonaws.services.cognitoidentity.AmazonCognitoIdentityClient.getId(AmazonCognitoIdentityClient.java:809)
        at com.amazonaws.auth.AWSAbstractCognitoIdentityProvider.getIdentityId(AWSAbstractCognitoIdentityProvider.java:172)
        at com.amazonaws.mobile.client.AWSMobileClientCognitoIdentityProvider.refresh(AWSMobileClient.java:3923)
        at com.amazonaws.auth.CognitoCredentialsProvider.retryRefresh(CognitoCredentialsProvider.java:723)
        at com.amazonaws.auth.CognitoCredentialsProvider.startSession(CognitoCredentialsProvider.java:695)
        at com.amazonaws.auth.CognitoCredentialsProvider.refresh(CognitoCredentialsProvider.java:640)
        at com.amazonaws.auth.CognitoCachingCredentialsProvider.refresh(CognitoCachingCredentialsProvider.java:511)
        at com.amazonaws.mobile.client.AWSMobileClient.federateWithCognitoIdentity(AWSMobileClient.java:1834)
        at com.amazonaws.mobile.client.AWSMobileClient$10.run(AWSMobileClient.java:1790)
        at com.amazonaws.mobile.client.internal.InternalCallback.await(InternalCallback.java:115) 
        at com.amazonaws.mobile.client.AWSMobileClient.federatedSignInWithoutAssigningState(AWSMobileClient.java:1731) 
        at com.amazonaws.mobile.client.AWSMobileClient$6$1.onSuccess(AWSMobileClient.java:1223) 
        at com.amazonaws.mobileconnectors.cognitoidentityprovider.CognitoUser.getSession(CognitoUser.java:1020) 
        at com.amazonaws.mobile.client.AWSMobileClient$6.run(AWSMobileClient.java:1208) 
        at com.amazonaws.mobile.client.internal.InternalCallback$1.run(InternalCallback.java:101) 
        at java.lang.Thread.run(Thread.java:923) `

I force close the app, Auth.fetchAuthSession() will return false again. 
Basically I cannot keep the user logged in.

**Environment(please complete the following information):**
implementation 'com.amplifyframework:aws-auth-cognito:1.18.0'
// Support for Java 8 features
coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.1.5'
implementation 'com.amplifyframework:rxbindings:1.18.0'

**Device Information (please complete the following information):**
 - Device: pixel 3
 - Android Version: 11

**Comment by richardmcclellan** [comment]
It sounds like you are not actually getting signed in successfully, due to this issue:

```
Caused by: com.amazonaws.AmazonServiceException: 1 validation error detected: Value 'eu-west-xxxxxxxxx' at 'identityPoolId' failed to satisfy constraint: Member must satisfy regular expression pattern: [\w-]+:[0-9a-f-]+ (Service: AmazonCognitoIdentity; Status Code: 400; Error Code: ValidationException; Request ID: 9bef8fcd-5fb2-45ce-b160-8ed8954ca8e2)
```

It looks like you are using a user pool id in a location where an identity pool id is expected.

Can you provide your `amplifyconfiguration.json` (with sensitive values redacted) to help debug further?

**Comment by mariuspena** [comment]
hey @richardfung this is my amplifyconfiguration.json

`
{
  "auth": {
    "plugins": {
      "awsCognitoAuthPlugin": {
        "IdentityManager": {
          "Default": {}
        },
        "CredentialsProvider": {
          "CognitoIdentity": {
            "Default": {
              "PoolId": "eu-west-1_xxxxxxxx",
              "Region": "eu-west-1"
            }
          }
        },
        "CognitoUserPool": {
          "Default": {
            "PoolId": "eu-west-1_xxxxxxxx",
            "AppClientId": "zzzzzzzzzzzzzzzzzzzzzz",
            "Region": "eu-west-1"
          }
        },
        "Auth": {
          "Default": {
            "authenticationFlowType": "USER_PASSWORD_AUTH"
          }
        }
      }
    }
  }
}
`

**Comment by mariuspena** [comment]
I guess I do not have the correct pool id under CognitoIdentity.
Documentation states I need something like this 
us-west-1:**123e4567-e89b-12d3-a456-426614174000**
but my ^ bold 2nd part is not that long

**Comment by richardmcclellan** [comment]
> I guess I do not have the correct pool id under CognitoIdentity.
> Documentation states I need something like this
> us-west-1:**123e4567-e89b-12d3-a456-426614174000**
> but my ^ bold 2nd part is not that long

Right.  Under `CredentialsProvider.CognitoIdentity.Default.PoolId`, you should put your Identity Pool ID, but what you have looks like a User Pool ID.

If possible, I'd suggest using the Amplify CLI, since it's a little less error prone than manually constructing the `amplifyconfiguration.json`.  You can simply run `amplify add auth` to create a new User Pool and Identity Pool.  Or, if you have existing resources, you can import them by running `amplify import auth`.

**Comment by mariuspena** [comment]
@richardmcclellan my javascript colleague uses a configuration file without cognito identity pool, just the branch with cognito user pool. 

I used amplify cli (amplify pull .... etc etc) and I got to a point where I get this error: 
`Failed to pull the backend.
No Amplify apps found. Please ensure your local profile matches the AWS account or region in which the Amplify app exists.
Error: No Amplify apps found. Please ensure your local profile matches the AWS account or region in which the Amplify app exists.
`
